### ISSUE TITLE
AuthUI.IdpConfig.EmailBuilder() InflationException Error

### ISSUE THREAD

**Issue by whittenator** [firebase/FirebaseUI-Android#1413]
### Step 2:
  * Android device: Pixel 2 XL
  * Android OS version: API 28 and API 27(Oreo)
  * Google Play Services version: 4.0.0
  * Firebase/Play Services SDK version: 16.0.1
  * FirebaseUI version: 4.1.0
  
### Step 3: Describe the problem:
Gmail and Phone buttons work. But, when the Email Signup button is clicked on the following error occurs for API 27(Oreo):
_com.firebase.ui.auth.FirebaseUiException: Code: 13, message: 13:_ 

and for API 28:
_android.view.InflateException: Binary XML file line #24: Binary XML file line #24: Error inflating class android.support.design.widget.TextInputLayout
    Caused by: android.view.InflateException: Binary XML file line #24: Error inflating class android.support.design.widget.TextInputLayout
    Caused by: java.lang.reflect.InvocationTargetException_

#### Steps to reproduce:

  1. Basic AuthUI.getInstance
 
  
#### Observed Results:

Error with app crash occurs.
  
#### Expected Results:

  Email sign in activity is presented.
  
#### Relevant Code:

  ```
  FirebaseAuth auth = FirebaseAuth.getInstance();
        if (auth.getCurrentUser() != null) {
            // already signed in
        } else {
            // not signed in
            startActivityForResult(
                    // Get an instance of AuthUI based on the default app
                    AuthUI.getInstance()
                            .createSignInIntentBuilder()
                    .setAvailableProviders(Arrays.asList(
                    new AuthUI.IdpConfig.GoogleBuilder().build(),
                    new AuthUI.IdpConfig.EmailBuilder().build(),
                    new AuthUI.IdpConfig.PhoneBuilder().build()))
                    .build(),
                    RC_SIGN_IN);
        }
  ```
#### ERROR API 28
```
android.view.InflateException: Binary XML file line #24: Binary XML file line #24: Error inflating class android.support.design.widget.TextInputLayout
    Caused by: android.view.InflateException: Binary XML file line #24: Error inflating class android.support.design.widget.TextInputLayout
    Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Constructor.newInstance0(Native Method)
        at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
        at android.view.LayoutInflater.createView(LayoutInflater.java:647)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:790)
        at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:730)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:863)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:866)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.rInflate(LayoutInflater.java:866)
        at android.view.LayoutInflater.rInflateChildren(LayoutInflater.java:824)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:515)
        at android.view.LayoutInflater.inflate(LayoutInflater.java:423)
        at com.firebase.ui.auth.ui.email.CheckEmailFragment.onCreateView(CheckEmailFragment.java:89)
        at android.support.v4.app.Fragment.performCreateView(Fragment.java:2442)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1460)
        at android.support.v4.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1784)
        at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1852)
        at android.support.v4.app.BackStackRecord.executeOps(BackStackRecord.java:802)
        at android.support.v4.app.FragmentManagerImpl.executeOps(FragmentManager.java:2625)
        at android.support.v4.app.FragmentManagerImpl.executeOpsTogether(FragmentManager.java:2411)
        at android.support.v4.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManager.java:2366)
        at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:2273)
        at android.support.v4.app.FragmentManagerImpl.dispatchStateChange(FragmentManager.java:3273)
        at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:3229)
        at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:201)
        at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:620)
        at android.support.v7.app.AppCompatActivity.onStart(AppCompatActivity.java:178)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1391)
        at android.app.Activity.performStart(Activity.java:7157)
        at android.app.ActivityThread.handleStartActivity(ActivityThread.java:2937)
        at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:180)
        at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:165)
        at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:142)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:70)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1808)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6669)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
    Caused by: java.lang.NoSuchFieldError: No static field error_color_material of type I in class Landroid/support/v7/appcompat/R$color; or its superclasses (declaration of 'android.support.v7.appcompat.R$color' appears in /data/app/com.example.traviswhitten.whatspoppin-tsoJTg75SeXYhS_YN1iu8A==/split_lib_slice_4_apk.apk)
        at android.support.design.widget.TextInputLayout.setErrorEnabled(TextInputLayout.java:654)
        at android.support.design.widget.TextInputLayout.<init>(TextInputLayout.java:257)
        at android.support.design.widget.TextInputLayout.<init>(TextInputLayout.java:187)
        	... 42 more
```

**Comment by samtstern** [comment]
@whittenator do you mind showing your dependencies?

**Comment by mbombonato** [comment]
Hello I'm facing the same issue after update from 3.x to 4.x
Tried some stuff here without success...

My dependencies are:
dependencies {
    implementation fileTree(include: ['*.jar'], dir: 'libs')
    implementation 'com.android.support:appcompat-v7:28.0.0-rc01'

    implementation 'com.android.support:support-v4:28.0.0-rc01'
    implementation 'com.android.support:recyclerview-v7:28.0.0-rc01'
    implementation 'com.android.support:cardview-v7:28.0.0-rc01'
    implementation 'com.android.support:exifinterface:28.0.0-rc01'
    implementation 'com.android.support.constraint:constraint-layout:1.1.2'

    implementation 'com.google.android.gms:play-services-maps:15.0.1'
    implementation 'com.google.android.gms:play-services-location:15.0.1'
    implementation 'com.google.firebase:firebase-core:16.0.1'
    implementation 'com.firebaseui:firebase-ui-auth:4.1.0'

    implementation 'com.facebook.android:facebook-android-sdk:4.32.0'
    implementation 'com.google.maps.android:android-maps-utils:0.5'
    implementation 'com.android.volley:volley:1.1.0'
    implementation 'com.redmadrobot:inputmask:3.4.4'
    implementation 'com.facebook.fresco:fresco:1.9.0'

    implementation 'org.jetbrains.kotlin:kotlin-stdlib:1.2.61'
    implementation 'com.android.support:multidex:1.0.3'
}

**Comment by mbombonato** [comment]
To make it work again, I reversed to com.android.support:support-v4:27.1.1

**Comment by DeerSteak** [comment]
Any idea when this might be fixed? This appears to still be an issue. I created a new project with Firebase today with a target of 28, and the Email button still crashes. Tons of searching finally led me to this issue. I reset everything back to 27.1.1 and set the compileSdkVersion and targetSdkVersion back to 27, and it's fine.

**Comment by raid5** [comment]
I'm still seeing this issue with v28 of the support libs, which was released on September 21st, 2018. Reverting back to 27.1.1 does indeed fix the issue, but when can we expect this to be fixed for v28?

**Comment by ShaldorZa** [comment]
This Problem is still currently an issue as of 17.11.2018

**Comment by samtstern** [comment]
I was recently able to update FirebaseUI (on the `version-4.3.0-dev` branch) to use the support libs version `28.0.0` without issue.

Is anyone on this thread willing to test out our `4.3.0-SNAPSHOT` build to see if it's fixed for you?
https://github.com/firebase/FirebaseUI-Android#snapshot-builds

**Comment by cutiko** [comment]
Greetings

According to this [SO answer](https://stackoverflow.com/a/52292323/4017501), we should add the design library. Did it and **the stable version will not crash anymore**.

```
implementation 'com.android.support:design:28.0.0'
```

Try the `4.3.0-SNAPSHOT` without the design library and the email button is not crashing. 
Try the `4.3.0-SNAPSHOT` using the design library and I'm not able to compile.

**Comment by samtstern** [comment]
@cutiko that's surprising since the `4.3.0-SNAPSHOT` library depends on the `28.0.0` design library.  What's the compile issue you get?

**Comment by samtstern** [comment]
I wonder if this would be fixed if we changed our dependency on `support:design` to be `api` not `implementation`

**Comment by cutiko** [comment]
@samtstern I see that my previous answers could have caused a confusion, just in case I edit it. 

Want to confirm that `4.3.0-SNAPSHOT` **fix the issue**. Apologies for any misunderstanding.

**Comment by samtstern** [comment]
Thanks for confirming @cutiko

**Comment by samtstern** [comment]
This has been released in version `4.3.0`.
